File: automator/src/main/java/com/stardust/automator/ActionArgument.java
package com.stardust.automator;

import android.os.Bundle;

/**
 * Created by Stardust on 2017/3/9.
 */

public abstract class ActionArgument {

    protected final String mKey;

    private ActionArgument(String key) {
        mKey = key;
    }

    public abstract void putIn(Bundle bundle);

    public static class IntActionArgument extends ActionArgument {

        private final int mInt;

        public IntActionArgument(String name, int i) {
            super(name);
            mInt = i;
        }

        @Override
        public void putIn(Bundle bundle) {
            bundle.putInt(mKey, mInt);
        }
    }

    public static class CharSequenceActionArgument extends ActionArgument {

        private final CharSequence mCharSequence;

        public CharSequenceActionArgument(String name, CharSequence charSequence) {
            super(name);
            mCharSequence = charSequence;
        }

        @Override
        public void putIn(Bundle bundle) {
            bundle.putCharSequence(mKey, mCharSequence);
        }
    }

    public static class FloatActionArgument extends ActionArgument {
        private final float mFloat;

        public FloatActionArgument(String name, float value) {
            super(name);
            mFloat = value;
        }

        @Override
        public void putIn(Bundle bundle) {
            bundle.putFloat(mKey, mFloat);
        }
    }

}

File: automator/src/main/java/com/stardust/automator/ActionArgument.kt
package com.stardust.automator

import android.os.Bundle

/**
 * Created by Stardust on 2017/3/9.
 */

abstract class ActionArgument private constructor(protected val mKey: String) {

    abstract fun putIn(bundle: Bundle)

    class IntActionArgument(name: String, private val mInt: Int) : ActionArgument(name) {

        override fun putIn(bundle: Bundle) {
            bundle.putInt(mKey, mInt)
        }
    }

    class CharSequenceActionArgument(name: String, private val mCharSequence: CharSequence) : ActionArgument(name) {

        override fun putIn(bundle: Bundle) {
            bundle.putCharSequence(mKey, mCharSequence)
        }
    }

    class FloatActionArgument(name: String, private val mFloat: Float) : ActionArgument(name) {

        override fun putIn(bundle: Bundle) {
            bundle.putFloat(mKey, mFloat)
        }
    }

}
